src/Microsoft.VisualStudio.Extensibility.Testing.SourceGenerator.UnitTests/Resources/TestGenerationForVS2012/WorkspaceInProcess1.g.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.VisualStudio.Extensibility.Testing
+{
+    [TestService]
```

Comment:
I'm not super up on patterns for partial classes, is there a reason we shouldn't put this attribute on the part of the partial that defines the constructor? Or is that somehow harder to code-gen?

Reply:
It's a chicken/egg situation. The constructor is generated because this attribute was already located.